clloader/scenario_loaders/classIncremental.py

```diff
@@ -0,0 +1,200 @@
+
+
+
+from typing import Callable, List, Tuple, Union
+
+import numpy as np
+import torch
+from clloader.datasets import BaseDataset, ContinuumDataset
+from torch.utils.data import Dataset as TorchDataset
+from torchvision import transforms
+from loader import CLLoader
+
+############
+# IDEE: Avoir une classe CLLoader de base et plusieur autres spécifiques au scenarii
+
+class ClassIncremental(CLLoader):
+    """Continual Loader, generating datasets for the consecutive tasks.
+
+    :param cl_dataset: A continual dataset.
+    :param increment: Either number of classes per task, or a list specifying for
+                      every task the amount of new classes.
+    :param initial_increment: A different task size applied only for the first task.
+                              Desactivated if `increment` is a list.
+    :param train_transformations: A list of data augmentation applied to the train set.
+    :param common_transformations: A list of transformations applied to both the
+                                   the train set and the test set. i.e. normalization,
+                                   resizing, etc.
+    :param evaluate_on: How to evaluate on val/test, either on all `seen` classes,
+                        on the `current` classes, or on `all` classes.
+    :param class_order: An optional custom class order, used for NC.
+    """
+
+    def __init__(
```

Comment:
Do we want some kind of abstract scenarii defining common methods?

Then all subsequents scenario override only a few method, similar to what was done for datasets.

Reply:
Oh, was that your "classe CLLoader de base"? If so, nevermind.

Reply:
yes, this is the idea, CLLoader could be the scenarii abstract class